components/common/src/command/package/install.rs

```diff
@@ -405,9 +406,13 @@ fn run_install_hook<T>(ui: &mut T, package: &PackageInstall) -> Result<()>
         ui.status(Status::Executing,
                   format!("install hook for '{}'", &package.ident(),))?;
         templating::compile_for_package_install(package)?;
-        if !hook.run(&package.ident().name,
-                     &Pkg::from_install(package)?,
-                     None::<&str>)
+        let mut pkg = Pkg::from_install(package)?;
+        // Install hooks do not have access to svc_passwords so
+        // we execute them under the current user account.
+        if let Some(user) = users::get_current_username() {
```

Comment:
What would be the situation where the user doesn't have a username (besides some interesting container scenarios I can think of)?

What kind of feedback would we get if there was no username, but the `pkg.svc_user` was somehow incompatible with this install use case?

Since service passwords are only a thing on Windows, can we make this change Windows-only?

Reply:
I honestly cannot imagine there being no username. The only way this could happen is if someone removed the USERNAME environment variable which could wreak all sorts of havoc throughout their system. I'm just trying to avoid the unwrap/expect. There is another issue that will use an API instead of the environment variable that will avoid the environment variable alltogether.